shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -922,7 +922,8 @@ SemanticsObject hitTest(float[] point) {
                 return null;
             if (childrenInHitTestOrder != null) {
                 final float[] transformedPoint = new float[4];
-                for (int i = 0; i < childrenInHitTestOrder.size(); i += 1) {
+                int size = childrenInHitTestOrder.size();
```

Comment:
This makes me think that `childrenInHitTestOrder` is actually not in hit test order? Do we need to update that and the documentation around it?

Reply:
I think you are right, the bug is in the framework side:

I/flutter ( 2814): hitTest: [1]
I/flutter ( 2814): traversal: [1]
I/flutter ( 2814): hitTest: [2]
I/flutter ( 2814): traversal: [2]
I/flutter ( 2814): hitTest: [3, 4, 6, 7]
I/flutter ( 2814): traversal: [3, 4, 6, 7]
I/flutter ( 2814): hitTest: [5]
I/flutter ( 2814): traversal: [5]
I/flutter ( 2814): hitTest: [8]
I/flutter ( 2814): traversal: [8]
I/flutter ( 2814): hitTest: [23]
I/flutter ( 2814): traversal: [23]
I/flutter ( 2814): hitTest: [9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22]
I/flutter ( 2814): traversal: [9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22]

Reply:
I'm pretty sure this field is the rename of the old `children` property, whose value used to come (and still does) from `SemanticsNode._children` property. IOW, if it's a bug, it's not a new bug.

Reply:
Spotted the bug

Reply:
This doesn't invert anything, it just copies
```
      // _children is sorted in paint order, so we invert it to get the hit test
      // order.
      childrenInHitTestOrder = new Int32List(childCount);
      for (int i = childCount - 1; i >= 0; i -= 1) {
        childrenInHitTestOrder[i] = _children[i].id;
      }
```

Should be like this:

```
      childrenInHitTestOrder = new Int32List(childCount);
      for (int i = childCount - 1; i >= 0; i -= 1) {
        childrenInHitTestOrder[i] = _children[childCount - i - 1].id;
      }
```

Reply:
Ah! Good find.

Reply:
for-loops, amirite?

Reply:
D'oh! :man_facepalming: